hw/ip_templates/racl_ctrl/rtl/racl_ctrl.sv.tpl

```diff
@@ -138,61 +134,95 @@ module ${module_instance_name} import ${module_instance_name}_reg_pkg::*; #(
+  racl_error_log_t [NumAllIps-1:0] combined_racl_error;
+
+  // Combine the internal and external RACL log to a single valid vector (for assertion) and a
+  // combined error vector for common handling in the logging logic.
   always_comb begin
-    // Default to the racl_ctrl reg_top error information. Possible since only
-    // one error allowed at a time.
-    racl_error_role        = racl_ctrl_racl_error_log.racl_role;
-    racl_error_ctn_uid     = racl_ctrl_racl_error_log.ctn_uid;
-    racl_error_read_access = racl_ctrl_racl_error_log.read_access;
-    for (int unsigned i = 0; i < (NumSubscribingIps + NumExternalSubscribingIps); i++) begin
-      racl_error_role        |= combined_racl_error_log[i].racl_role;
-      racl_error_ctn_uid     |= combined_racl_error_log[i].ctn_uid;
-      racl_error_read_access |= combined_racl_error_log[i].read_access;
+    for (int unsigned i = 0; i < NumSubscribingIps; i++) begin
+      combined_racl_error_valid[i] = racl_error_i[i].valid;
+      combined_racl_error[i]       = racl_error_i[i];
+    end
+
+    for (int unsigned i = NumSubscribingIps;
+         i < NumSubscribingIps + NumExternalSubscribingIps; i++) begin
+      combined_racl_error_valid[i] = racl_error_external_i[i - NumSubscribingIps].valid;
+      combined_racl_error[i]       = racl_error_external_i[i - NumSubscribingIps];
     end
+
+    // Last element is the internal RACL error of the own reg_top
+    combined_racl_error_valid[NumAllIps-1] = racl_ctrl_racl_error.valid;
+    combined_racl_error      [NumAllIps-1] = racl_ctrl_racl_error;
   end
 
+  // If there a multiple errors in the same cycle, arbitrate and get the first error.
+  // Lower index in combined_racl_error_valid has higher priority
+  logic [$clog2(NumAllIps)-1:0] error_idx;
+  prim_arbiter_fixed #(
+    .N          ( NumAllIps ),
+    .EnDataPort ( 0         )
+  ) u_prim_err_arb (
+    .clk_i,
+    .rst_ni,
+    .req_i    ( combined_racl_error_valid ),
+    .data_i   ( '{default:'0}             ),
+    .gnt_o    (                           ),
+    .idx_o    ( error_idx                 ),
+    .valid_o  (                           ),
+    .data_o   (                           ),
+    .ready_i  ( 1'b1                      )
+  );
+  racl_error_log_t racl_error_arb;
+  assign racl_error_arb = combined_racl_error[error_idx];
+
+  // If there are multiple errors at the same time, we need to directly assert the
+  // overflow bit in the log. We use a one-hot checker to determine if there is more
+  // than one error. The checker asserts an error in these cases.
+  logic multiple_errors;
+  prim_onehot_check #(
+    .OneHotWidth ( NumAllIps ),
+    .AddrCheck   ( 0         ),
+    .EnableCheck ( 1         ),
+    .StrictCheck ( 0         )
+  ) u_multiple_err_det (
+    .clk_i,
+    .rst_ni,
+    .oh_i   ( combined_racl_error_valid ),
+    .addr_i ( '0                        ),
+    .en_i   ( 1'b1                      ),
+    .err_o  ( multiple_errors           )
```

Comment:
This does not seem to be working that way. It's assumed that the error output triggers an alert by using the `ASSERT_PRIM_COUNT_ERROR_TRIGGER_ALERT` assertion macro. Is there a similar prim available without that alert condition?

Reply:
Any reason you can't just use the following?
```
assign multiple_error = racl_error_arb.valid & |(combined_racl_error & ~('b1 << error_idx));
```

Reply:
I've successfully simulated these changes (_with_ the arbiter-based implementation of 'multiple_error', as discussed) so I'm happy for this to be merged once that's applied and the linting is clean.